utils/terrascript_client.py

```diff
@@ -56,37 +59,72 @@ def __init__(self, integration, integration_prefix, oc_map):
                         region=config['region'])
             ts += terraform(backend=b)
             tss[name] = ts
+            locks[name] = Lock()
         self.tss = tss
+        self.locks = locks
 
     def populate_configs_and_vars_from_vault(self):
         self.init_accounts()
-        self.configs = self.get_vault_tf_secrets('config')
-        self.variables = self.get_vault_tf_secrets('variables')
+
+        vault_specs = self.init_vault_tf_secret_specs()
+        pool = ThreadPool(self.thread_pool_size)
+        results = pool.map(self.get_vault_tf_secrets, vault_specs)
+
+        self.configs = {}
+        self.variables = {}
+        for account_name, type, secret in results:
+            if type == 'config':
+                self.configs[account_name] = secret
+            if type == 'variables':
+                self.variables[account_name] = secret
 
     def init_accounts(self):
         config = get_config()
         accounts = config['terraform']
         self.accounts = accounts.items()
 
-    def get_vault_tf_secrets(self, type):
-        secrets = {}
-        for name, data in self.accounts:
-            secrets_path = data['secrets_path']
-            secret = vault_client.read_all(secrets_path + '/' + type)
-            secrets[name] = secret
-        return secrets
+    def init_vault_tf_secret_specs(self):
+        vault_specs = []
+        for account_name, data in self.accounts:
+            for type in ('config', 'variables'):
+                init_spec = {'account_name': account_name,
+                             'data': data,
+                             'type': type}
+                vault_specs.append(init_spec)
+        return vault_specs
+
+    def get_vault_tf_secrets(self, init_spec):
+        account = init_spec['account']
+        data = init_spec['data']
+        type = init_spec['type']
+        secrets_path = data['secrets_path']
+        secret = vault_client.read_all(secrets_path + '/' + type)
+        return (account, type, secret)
 
     def populate(self, tf_query):
+        populate_specs = self.init_populate_specs(tf_query)
+
+        pool = ThreadPool(self.thread_pool_size)
+        pool.map(self.populate_tf_resources, populate_specs)
+
+        self.validate()
+
+    def init_populate_specs(self, tf_query):
```

Comment:
all these functions are better defined as list comprehensions

Reply:
on second thought these are complex functions. It's clear what they do, maybe a list comprehension is not necessary here.